backend/daemon/api/documents/v1alpha/comments.go

```diff
@@ -0,0 +1,343 @@
+			if err != nil {
+				return errutil.ParseError("selectors.blockRevision", sel.BlockRevision, rev, err)
+			}
+			batch.Add(snode, mttdoc.AttrConvSelectorBlockID, sel.BlockId)
+			batch.Add(snode, mttdoc.AttrConvSelectorBlockRevision, rev)
+			// If we have end on the selector we must have start, meaning that it's
+			// a fine-grained selection, not the whole block.
+			if sel.End != 0 {
+				batch.Add(snode, mttdoc.AttrConvSelectorStart, int(sel.Start))
+				batch.Add(snode, mttdoc.AttrConvSelectorEnd, int(sel.End))
+			}
+			batch.Add(vcs.RootNode, mttdoc.AttrConvSelector, snode)
+		}
+
+		batch.Add(vcs.RootNode, mttdoc.AttrBlockSnapshot, encodeBlock(in.InitialComment))
+
+		dirty := batch.Dirty()
+
+		conn.AddDatoms(convo, change, dirty...)
+		conn.SaveVersion(convo, "main", meLocal, vcsdb.LocalVersion{change})
+		conn.EncodeChange(change, me.DeviceKey())
+
+		for _, d := range dirty {
+			if err := backlinks.IndexDatom(conn, convo, change, d); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return api.loadConversation(conn, perma.ID)
+}
+
+// AddComment implements the Comments server.
+func (api *Server) AddComment(ctx context.Context, in *documents.AddCommentRequest) (*documents.Block, error) {
+	me, err := api.me.Await(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if in.ConversationId == "" {
+		return nil, errutil.MissingArgument("conversationID")
+	}
+
+	convid, err := cid.Decode(in.ConversationId)
+	if err != nil {
+		return nil, errutil.ParseError("conversationID", in.ConversationId, convid, err)
+	}
+
+	if err := validateComment("comment", in.Comment); err != nil {
+		return nil, err
+	}
+
+	conn, release, err := api.vcsdb.Conn(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+
+	if err := conn.WithTx(true, func() error {
+		meLocal := conn.LookupIdentity(me)
+		obj := conn.LookupPermanode(convid)
+
+		clock := hlc.NewClock()
+		main := conn.GetVersion(obj, "main", meLocal)
+		for _, v := range main {
+			clock.Track(hlc.Unpack(conn.GetChangeMaxTime(obj, v)))
+		}
+
+		change := conn.NewChange(obj, meLocal, main, clock)
+		batch := vcs.NewBatch(clock, me.DeviceKey().Abbrev())
+
+		batch.Add(vcs.RootNode, mttdoc.AttrBlockSnapshot, encodeBlock(in.Comment))
+
+		dirty := batch.Dirty()
+		conn.AddDatoms(obj, change, dirty...)
+		conn.SaveVersion(obj, "main", meLocal, vcsdb.LocalVersion{change})
+		blk := conn.EncodeChange(change, me.DeviceKey())
+
+		if err := conn.Err(); err != nil {
+			return nil
+		}
+
+		in.Comment.Revision = blk.Cid().String()
+
+		for _, d := range dirty {
+			if err := backlinks.IndexDatom(conn, obj, change, d); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return in.Comment, nil
+}
+
+// DeleteConversation implements the Comments server.
+func (api *Server) DeleteConversation(context.Context, *documents.DeleteConversationRequest) (*emptypb.Empty, error) {
```

Comment:
Humm, have we spoken about deleting conversations? What about a link pointing to a deleted conversation? or transclusion? Isn't it the same as deleting a document, which is something we don't allow?

Reply:
A conversation is like a block of a document, you can indeed delete a block of a document and publish a new version.

Reply:
> Humm, have we spoken about deleting conversations? What about a link pointing to a deleted conversation? or transclusion? Isn't it the same as deleting a document, which is something we don't allow?

It's not implemented currently. The concerns you have are all valid, and are one of the reasons it's left unimplemented. But as a user I'd like to delete conversations, or resolve them somehow to get them out of my way. As they are independent objects there's nothing preventing us from removing them. But we indeed have to be careful when other links to this object being delete exist elsewhere.

> A conversation is like a block of a document, you can indeed delete a block of a document and publish a new version.

No, a Conversation is like another document pointing to the original document.

Reply:
Ok, I guess it can be left unimplemented until we have a clearer understanding of the implications.